pandas/indexes/base.py

```diff
@@ -2886,6 +2888,9 @@ def join(self, other, how='left', level=None, return_indexers=False):
         elif how == 'outer':
             join_index = self.union(other)
 
+        if sort:
```

Comment:
No, sort=True is only _ignored_ for outer join, because the outer join always sorts the result index. For the other types of join (left, right, inner), the default result is unsorted, as it preserves the left index order (for left and inner) or the right index order (for right), which might not be previously sorted.

By setting sort=True, we force the resulting index order to be sorted, even for the cases of left, right and inner joins.

What do you mean by _I would set it False_? The parameter sort is set False by default.